### PR TITLE
Add ConsumerName option, use similar ephemeral names as server

### PR DESCRIPTION
**Pull request by wallyqs**

### CODE REVIEW

File: js.go [hunk 1]
```diff
@@ -2474,6 +2476,14 @@ func ConsumerMemoryStorage() SubOpt {
 	})
 }
 
+// ConsumerName sets the name for a consumer.
```
Comment (derekcollison) [inline js.go:2479]: If I use this and set it, it will still use long form create yes?

File: js.go [hunk 2]
```diff
@@ -3646,3 +3656,22 @@ func (st *StorageType) UnmarshalJSON(data []byte) error {
 	}
 	return nil
 }
+
+// Length of our hash used for named consumers.
+const nameHashLen = 8
+
+// Computes a hash of 8 characters for the name.
+func getHash(name string) string {
```
Comment (derekcollison) [inline js.go:3664]: Since we don't use any other size would collapse into one function with fixed size of 8 for now.

Comment (wallyqs) [reply js.go:3664]: sounds good

Comment (wallyqs) [reply js.go:3664]: updated to a single function

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**